### PR TITLE
avatar unit tests

### PR DESCRIPTION
**Pull request by brianchristopherbrady**
Adds Avatar unit tests

### CODE REVIEW

**Comment by fabricteam** [comment]
## 🕵 fluentui-web-components-v3 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_web-components"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/c83bfec16fd0d93b1305e5f5aec5d8a62c8f7cb7'>c83bfec16fd0d93b1305e5f5aec5d8a62c8f7cb7</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/web-components/src/avatar/avatar.spec.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  test('should render with custom initials based on the provided initials value', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+        <fluent-avatar initials="JD"></fluent-avatar>
+      `;
+    });
+
+    await expect(element).toHaveText('JD');
+  });
```
Comment (chrisdholt) [inline packages/web-components/src/avatar/avatar.spec.ts:46]: I don't think this actually fully tests what we'd like to see. This does provide initials when initials are provided, but I think we should validate that something like a name of `Julie Wright` with custom initials provided of `JJ` would yeild, `JJ`.

Comment (chrisdholt) [reply packages/web-components/src/avatar/avatar.spec.ts:46]: Both are probably worthwile but we should be specific, "When no name value is set" for the first, "When name value is set" for the second.

File: packages/web-components/src/avatar/avatar.spec.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+  test('should set the size attribute to the provided size value', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+        <fluent-avatar size="48"></fluent-avatar>
+      `;
+    });
+
+    await expect(element).toHaveAttribute('size', '48');
+  });
```
Comment (chrisdholt) [inline packages/web-components/src/avatar/avatar.spec.ts:56]: Should we test all? What about when size is _updated_, etc?

Comment (chrisdholt) [reply packages/web-components/src/avatar/avatar.spec.ts:56]: Same for the below tests

File: packages/web-components/src/avatar/avatar.spec.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  test('should set the color attrinute to the provided color value', async () => {
+    await root.evaluate(node => {
+      node.innerHTML = /* html */ `
+        <fluent-avatar color="brand"></fluent-avatar>
+      `;
+    });
+
+    await expect(element).toHaveAttribute('color', 'brand');
+  });
+});
```
Comment (chrisdholt) [inline packages/web-components/src/avatar/avatar.spec.ts:87]: What about `active`, using `colorId`, the `this.color` colorful portion (should that be tested for input/output)?

**Review by chrisdholt — COMMENTED** [review]
I think we need to get a bit more robust on the tests here.

File: packages/web-components/src/avatar/avatar.spec.ts (hunk 3 above)
Comment (brianchristopherbrady) [reply packages/web-components/src/avatar/avatar.spec.ts:204]: I was having a heck of a time testing the color-id scenarios. Let me know what you think.

**Review by halerankin — APPROVED** [review]

**Review by miroslavstastny — APPROVED** [review]

**Pull request merged**